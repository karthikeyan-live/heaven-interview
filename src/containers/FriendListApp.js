import React, { Component } from 'react';
import styles from './FriendListApp.css';
import { connect } from 'react-redux';

import { addFriend, deleteFriend, starFriend, toggleFriendSex } from '../actions/FriendsActions';
import { changeActivePage } from '../actions/FriendListActions';
import { FRIEND_LIST_ITEMS_COUNT_PER_PAGE as itemsCountPerPage } from '../constants/ActionTypes';
import { FriendList, AddFriendInput, FriendListPaginate } from '../components';

class FriendListApp extends Component {

  render() {
    const { friendlist: { friendsById } } = this.props;
    const { friendlist: { activePageOfFriendList } } = this.props;

    const actions = {
      addFriend: this.props.addFriend,
      deleteFriend: this.props.deleteFriend,
      starFriend: this.props.starFriend,
      toggleFriendSex: this.props.toggleFriendSex,
      changeActivePage: this.props.changeActivePage
    };

    const paging = {
      activePage: activePageOfFriendList,
      itemsCountPerPage: itemsCountPerPage,
      totalItemsCount: friendsById.length
    };

    return (
      <div className={styles.friendListApp}>
        <h1>The FriendList</h1>
        <AddFriendInput addFriend={actions.addFriend} />
        <FriendList friends={friendsById} actions={actions}  {...paging} />
        <FriendListPaginate {...paging} changeActivePage={actions.changeActivePage} />
      </div>
    );
  }
}

function mapStateToProps(state) {
  return state
}

export default connect(mapStateToProps, {
  addFriend,
  deleteFriend,
  starFriend,
  toggleFriendSex,
  changeActivePage
})(FriendListApp)
