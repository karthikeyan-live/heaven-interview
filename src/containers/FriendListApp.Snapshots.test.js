import React from 'react';
import renderer from 'react-test-renderer';
import friendsById from '../../tools/mockData';
import { FriendListApp } from './FriendListApp';

function renderFriendListApp(args) {
    let defaultProps = {
        friendlist: friendsById,
        itemsCountPerPage: 2,
        changeActivePage: jest.fn(),
        starFriend: jest.fn(),
        deleteFriend: jest.fn(),
        toggleFriendSex: jest.fn(),
        addFriend: jest.fn()
    };

    const props = { ...defaultProps, ...args };
    return renderer.create(<FriendListApp {...props} />)
}

describe('friend list app - connected component test', () => {

    it('render with mock data', () => {
        const component = renderFriendListApp();
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });

    it('navigate to 2nd page', () => {
        const component = renderFriendListApp({ friendlist: { ...friendsById, activePageOfFriendList: 2 } });
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });

    it('if friend list is empty legend should be "0 of 0"', () => {
        const component = renderFriendListApp({ friendlist: { ...friendsById, friendsById: [] } });
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
})