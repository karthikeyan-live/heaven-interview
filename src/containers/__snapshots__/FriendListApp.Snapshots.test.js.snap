exports[`friend list app - connected component test if friend list is empty legend should be "0 of 0" 1`] = `
<div
  className={undefined}>
  <h1>
    The FriendList
  </h1>
  <input
    autoFocus="true"
    className="form-control"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Type the name of a friend"
    type="text"
    value="" />
  <ul
    className={undefined} />
  <div
    className={undefined}>
    <div>
      <button
        className="btn btn-default undefined"
        disabled={true}
        onClick={[Function]}>
        <i
          className="fa fa-fast-backward" />
      </button>
      <button
        className="btn btn-default undefined"
        disabled={true}
        onClick={[Function]}>
        <i
          className="fa fa-backward" />
      </button>
    </div>
    <b>
      0
       of 
      0
    </b>
    <div>
      <button
        className="btn btn-default undefined"
        disabled={true}
        onClick={[Function]}>
        <i
          className="fa fa-forward" />
      </button>
      <button
        className="btn btn-default undefined"
        disabled={true}
        onClick={[Function]}>
        <i
          className="fa fa-fast-forward" />
      </button>
    </div>
  </div>
</div>
`;

exports[`friend list app - connected component test navigate to 2nd page 1`] = `
<div
  className={undefined}>
  <h1>
    The FriendList
  </h1>
  <input
    autoFocus="true"
    className="form-control"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Type the name of a friend"
    type="text"
    value="" />
  <ul
    className={undefined}>
    <li
      className={undefined}>
      <div
        className={undefined}>
        <div>
          <span>
            John F. Kennedy
             
            (M)
          </span>
        </div>
        <div>
          <small>
            xx friends in common
          </small>
        </div>
      </div>
      <div
        className={undefined}>
        <button
          className="btn btn-default undefined"
          onClick={[Function]}>
          <i
            className="fa fa-star-o" />
        </button>
        <button
          className="btn btn-default undefined"
          onClick={[Function]}>
          <i
            className="fa fa-trash" />
        </button>
        <button
          className="btn btn-default undefined"
          onClick={[Function]}>
          <i
            className="fa fa-male" />
        </button>
      </div>
    </li>
  </ul>
  <div
    className={undefined}>
    <div>
      <button
        className="btn btn-default undefined"
        disabled={false}
        onClick={[Function]}>
        <i
          className="fa fa-fast-backward" />
      </button>
      <button
        className="btn btn-default undefined"
        disabled={false}
        onClick={[Function]}>
        <i
          className="fa fa-backward" />
      </button>
    </div>
    <b>
      2
       of 
      2
    </b>
    <div>
      <button
        className="btn btn-default undefined"
        disabled={true}
        onClick={[Function]}>
        <i
          className="fa fa-forward" />
      </button>
      <button
        className="btn btn-default undefined"
        disabled={true}
        onClick={[Function]}>
        <i
          className="fa fa-fast-forward" />
      </button>
    </div>
  </div>
</div>
`;

exports[`friend list app - connected component test render with mock data 1`] = `
<div
  className={undefined}>
  <h1>
    The FriendList
  </h1>
  <input
    autoFocus="true"
    className="form-control"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Type the name of a friend"
    type="text"
    value="" />
  <ul
    className={undefined}>
    <li
      className={undefined}>
      <div
        className={undefined}>
        <div>
          <span>
            Ronald Reagan
             
            (M)
          </span>
        </div>
        <div>
          <small>
            xx friends in common
          </small>
        </div>
      </div>
      <div
        className={undefined}>
        <button
          className="btn btn-default undefined"
          onClick={[Function]}>
          <i
            className="fa fa-star" />
        </button>
        <button
          className="btn btn-default undefined"
          onClick={[Function]}>
          <i
            className="fa fa-trash" />
        </button>
        <button
          className="btn btn-default undefined"
          onClick={[Function]}>
          <i
            className="fa fa-male" />
        </button>
      </div>
    </li>
    <li
      className={undefined}>
      <div
        className={undefined}>
        <div>
          <span>
            Gerald Ford
             
            (M)
          </span>
        </div>
        <div>
          <small>
            xx friends in common
          </small>
        </div>
      </div>
      <div
        className={undefined}>
        <button
          className="btn btn-default undefined"
          onClick={[Function]}>
          <i
            className="fa fa-star-o" />
        </button>
        <button
          className="btn btn-default undefined"
          onClick={[Function]}>
          <i
            className="fa fa-trash" />
        </button>
        <button
          className="btn btn-default undefined"
          onClick={[Function]}>
          <i
            className="fa fa-male" />
        </button>
      </div>
    </li>
  </ul>
  <div
    className={undefined}>
    <div>
      <button
        className="btn btn-default undefined"
        disabled={true}
        onClick={[Function]}>
        <i
          className="fa fa-fast-backward" />
      </button>
      <button
        className="btn btn-default undefined"
        disabled={true}
        onClick={[Function]}>
        <i
          className="fa fa-backward" />
      </button>
    </div>
    <b>
      1
       of 
      2
    </b>
    <div>
      <button
        className="btn btn-default undefined"
        disabled={false}
        onClick={[Function]}>
        <i
          className="fa fa-forward" />
      </button>
      <button
        className="btn btn-default undefined"
        disabled={false}
        onClick={[Function]}>
        <i
          className="fa fa-fast-forward" />
      </button>
    </div>
  </div>
</div>
`;
